src/json-schema.js

```diff
@@ -134,13 +134,74 @@ function checkSchemaHasReferences(schema) {
   });
 }
 
+/** Traverses the entire schema to find all of the references
+ * @returns array of each reference that is found in the schema
+ */
+function findReferences(schema) {
+  if (schema.$ref) {
+    return [schema.$ref];
+  }
+
+  let references = [];
+
+  if (schema.allOf) {
+    references = references.concat(...schema.allOf.map(findReferences));
+  }
+
+  if (schema.anyOf) {
+    references = references.concat(...schema.anyOf.map(findReferences));
+  }
+
+  if (schema.oneOf) {
+    references = references.concat(...schema.oneOf.map(findReferences));
+  }
+
+  if (schema.not) {
+    references = references.concat(...findReferences(schema.not));
+  }
+
+  // Array
+
+  if (schema.items) {
+    if (Array.isArray(schema.items)) {
+      references = references.concat(...schema.items.map(findReferences));
+    } else {
+      references = references.concat(findReferences(schema.items));
+    }
+  }
+
+  if (schema.additionalItems && typeof schema.additionalItems === 'object') {
+    references = references.concat(findReferences(schema.additionalItems));
+  }
+
+  // Object
+
+  if (schema.properties) {
+    Object.keys(schema.properties).forEach((key) => {
+      references = references.concat(findReferences(schema.properties[key]));
+    });
+  }
+
+  if (schema.patternProperties) {
+    Object.keys(schema.patternProperties).forEach((key) => {
+      references = references.concat(findReferences(schema.patternProperties[key]));
+    });
+  }
+
+  if (schema.additionalProperties && typeof schema.additionalProperties === 'object') {
+    references = references.concat(findReferences(schema.additionalProperties));
+  }
+
+  return references;
+}
+
 /** Convert Swagger schema to JSON Schema
  */
-export default function convertSchema(schema, root, convertDefinitions = true) {
-  const references = [];
+export function convertSchema(schema, root, copyDefinitions = true) {
```

Comment:
Before root was the original Swagger document, instead now it is always the converted JSON Schema definitions, and thus the parameter is renamed to copyDefinitions, to copy the referenced definitions over to the converted schema.